### DISCUSSION TITLE
Lists with code blocks not continuing numbering.

### DISCUSSION THREAD

**Discussion by adambirds** [squidfunk/mkdocs-material#4886] [Q&A] [answered]
Here is my code:

````markdown
## Installation Steps

1. Clone or download the repository.

1. Create a Python virtual environment and install the dependencies.

```bash
python3 -m venv venv
source venv/bin/activate
pip install -r requirements.txt
```

1. Create a PostgreSQL database and user.

```bash
sudo -u postgres psql
CREATE DATABASE battleofmidgard;
CREATE USER battleofmidgard WITH PASSWORD 'password';
GRANT ALL PRIVILEGES ON DATABASE battleofmidgard TO battleofmidgard;
```

1. Make a copy of config.yml.example and rename it to config.yml.

```bash
cp config.yml.example config.yml
```

1. Edit config.yml and add in the database credentials and details.
````

Now the numbers should continue, but instead, restart like so:

![image](https://user-images.githubusercontent.com/13500254/213471640-b3a5ff11-be4f-4a26-8d4e-9ceaca4a305d.png)

I have tried indenting the code blocks, which numbers correctly but then the code blocks don't render correctly.

Any ideas on how to resolve?

**Comment by adambirds** [comment]
Okay so I solved by surrounded my code blocks like so:

```markdown
1. Create a PostgreSQL database and user.
    <pre><code class="language-bash">sudo -u postgres psql
    CREATE DATABASE battleofmidgard;
    CREATE USER battleofmidgard WITH PASSWORD 'password';
    GRANT ALL PRIVILEGES ON DATABASE battleofmidgard TO battleofmidgard;
    </code></pre>
```

This fixed it, so the issue is that if a code block is indented, the `<pre></pre>` tags as well as the language class aren't being added around the code tags. Would you like me to create a bug report for this?

**Reply by facelessuser** [reply]
You don't need to wrap the content in `<pre></pre>`, just indent the original fenced syntax that you had before.

**Comment by squidfunk** [comment, marked as answer]
As @facelessuser already pointed out, here is your example with indented blocks, which will preserve numbering:

```` markdown
## Installation Steps

1. Clone or download the repository.

1. Create a Python virtual environment and install the dependencies.

    ```bash
    python3 -m venv venv
    source venv/bin/activate
    pip install -r requirements.txt
    ```

1. Create a PostgreSQL database and user.

    ```bash
    sudo -u postgres psql
    CREATE DATABASE battleofmidgard;
    CREATE USER battleofmidgard WITH PASSWORD 'password';
    GRANT ALL PRIVILEGES ON DATABASE battleofmidgard TO battleofmidgard;
    ```

1. Make a copy of config.yml.example and rename it to config.yml.

    ```bash
    cp config.yml.example config.yml
    ```

1. Edit config.yml and add in the database credentials and details.
````

**Reply by adambirds** [reply]
I tried this, but if you look at my screenshot, it messes up the rendering of the code blocks.

**Reply by adambirds** [reply]
Oh i forgot to include the screnshot, here it is when indented, they don't render properly
![image](https://user-images.githubusercontent.com/13500254/213489055-075a6e6b-6102-4661-a03c-4b601712bf99.png)

**Reply by facelessuser** [reply]
Your markdown in the opening post does not show the code blocks indented under the lists. The @squidfunk posted does have indentation. Assuming you are using SuperFences, this does work as I use them in my own documentation using Material and SuperFences together: https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#nested-fence-format

**Reply by facelessuser** [reply]
If you are using Codehilite, please read the Material documentation that suggests using SuperFences: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/?h=

**Reply by adambirds** [reply]
Thanks @facelessuser. I was just using the default settings, didn't realise I had to enable an extra markdown plugin for it.

**Reply by facelessuser** [reply]
Python Markdown follows the old school spec for Markdown. Fenced code blocks were not part of that spec. It implements fenced code block via the `fenced_code` extension, but this plugin can only render code blocks at the root level (no indentation). SuperFences allows for nesting code blocks anywhere. So, don't use `fenced_code` and `codehilite`, but use `pymdownx.superfences`.
